addons/sale_project/tests/test_sale_project.py

```diff
@@ -1032,3 +1039,148 @@ def test_confirm_sale_order_on_project_creation(self):
+            sol_new_project.project_id.account_id,
+            "As the projects of `sol_task_in_template_project` and `sol_new_project` were generated, they share the same AA which was created after SO confirmation."
+        )
+
+    def test_global_project_service_takes_so_project_on_so_confirmation(self):
+        self.product_order_service2.project_id = False
+        so = self.env['sale.order'].create({'partner_id': self.partner.id})
+        sol_task_in_global_project, sol_new_project = self.env['sale.order.line'].create([
+            {'order_id': so.id, 'product_id': self.product_order_service2.id},
+            {'order_id': so.id, 'product_id': self.product_order_service3.id},
+        ])
+        so.action_confirm()
+        self.assertEqual(
+            so.project_id,
+            sol_new_project.project_id,
+            "The project of the SO should be set to the project that was generated by `sol_new_project` at SO confirmation."
+        )
+        self.assertEqual(
+            so.project_id,
+            sol_task_in_global_project.task_id.project_id,
+            "The project of the task of `sol_task_in_global_project` should be set to the project of the SO."
+        )
+
+    def test_global_project_service_takes_so_project_on_already_confirmed_so(self):
+        self.product_order_service2.project_id = False
+        so = self.env['sale.order'].create({'partner_id': self.partner.id})
+        so.action_confirm()
+        sol_task_in_global_project, sol_new_project = self.env['sale.order.line'].create([
+            {'order_id': so.id, 'product_id': self.product_order_service2.id},
+            {'order_id': so.id, 'product_id': self.product_order_service3.id},
+        ])
+        self.assertEqual(
+            so.project_id,
+            sol_new_project.project_id,
+            "The project of the SO should be set to the project that was generated by `sol_new_project` after adding the SOLs in batch to the SO."
+        )
+        self.assertEqual(
+            so.project_id,
+            sol_task_in_global_project.task_id.project_id,
+            "The project of the task of `sol_task_in_global_project` should be set to the project of the SO."
+        )
+
+    def test_global_project_service_no_so_project_error(self):
+        self.product_order_service2.project_id = False
+        so = self.env['sale.order'].create({
+            'partner_id': self.partner.id,
+            'order_line': [Command.create({'product_id': self.product_order_service2.id})],
+        })
+        with self.assertRaises(UserError, msg="The SOL has a product which creates a task on SO confirmation, but no project is configured on the product or SO."):
+            so.action_confirm()
+
+    def test_so_confirmation_in_batch(self):
+        so1, so2 = self.env['sale.order'].create([{
+            'partner_id': self.partner.id,
+            'order_line': [Command.create({'product_id': self.product_order_service3.id})],
+        } for _ in range(2)])
```

Comment:
```suggestion
        } for _dummy in range(2)])
```
to avoid overriding `_` translation function. 😇 

I know inside tests, we will probably never need to call the translation function but it is more a best practise. 🙂

Reply:
Didn't know about that :open_mouth: Nice catch!